bref

```diff
@@ -52,6 +52,104 @@ $app->command('init', function (SymfonyStyle $io) {
+    /*
+     * TODO Mirror the directory instead of recreating it from scratch every time
+     * Blocked by https://github.com/symfony/symfony/pull/26399
+     * In the meantime we destroy `.bref/output` completely every time which
+     * is not efficient.
+     */
+    $fs->remove('.bref/output');
+    $fs->mkdir('.bref/output');
+    $filesToCopy = new Finder;
+    $filesToCopy->in('.')
+        ->depth(0)
+        ->exclude('.bref') // avoid a recursive copy
+        ->ignoreDotFiles(false);
+    foreach ($filesToCopy as $fileToCopy) {
+        if (is_file($fileToCopy->getPathname())) {
+            $fs->copy($fileToCopy->getPathname(), '.bref/output/' . $fileToCopy->getFilename());
+        } else {
+            $fs->mirror($fileToCopy->getPathname(), '.bref/output/' . $fileToCopy->getFilename(), null, [
+                'copy_on_windows' => true, // Force to copy symlink content
+            ]);
+        }
+    }
+
+    // Cache PHP's binary in `.bref/bin/php` to avoid downloading it
+    // on every deploy.
+    /*
+     * TODO Allow choosing a PHP version instead of using directly the
+     * constant `PHP_TARGET_VERSION`. That could be done using the `.bref.yml`
+     * config file: there could be an option in that config, for example:
+     * php:
+     *     version: 7.2.2
+     */
+    if (!$fs->exists('.bref/bin/php/php-' . PHP_TARGET_VERSION . '.tar.gz')) {
+        $io->writeln('Downloading PHP in the `.bref/bin/` directory');
+        $fs->mkdir('.bref/bin/php');
+        $defaultUrl = 'https://s3.amazonaws.com/bref-php/bin/php-' . PHP_TARGET_VERSION . '.tar.gz';
+        /*
+         * TODO This option allows to customize the PHP binary used. It should be documented
+         * and probably moved to a dedicated option like:
+         * php:
+         *     url: 'https://s3.amazonaws.com/...'
+         */
+        $url = $projectConfig['php'] ?? $defaultUrl;
+        $commandRunner->run("curl -sSL $url -o .bref/bin/php/php-" . PHP_TARGET_VERSION . ".tar.gz");
+    }
+
+    $io->writeln('Installing the PHP binary');
+    $fs->mkdir('.bref/output/.bref/bin');
+    $commandRunner->run('tar -xzf .bref/bin/php/php-' . PHP_TARGET_VERSION . '.tar.gz -C .bref/output/.bref/bin');
+
+    $io->writeln('Installing `handler.js`');
+    $fs->copy(__DIR__ . '/template/handler.js', '.bref/output/handler.js');
+
+    $io->writeln('Installing composer dependencies');
+    $commandRunner->run('cd .bref/output && composer install --no-dev --classmap-authoritative --no-scripts');
+
+    /*
+     * TODO Edit the `serverless.yml` copy (in `.bref/output` to deploy these files:
+     * - bref.php
+     * - handler.js
+     * - .bref/**
+     */
+
+    // Run build hooks defined in .bref.yml
+    $buildHooks = $projectConfig['hooks']['build'] ?? [];
+    foreach ($buildHooks as $buildHook) {
+        $io->writeln('Running ' . $buildHook);
+        $commandRunner->run('cd .bref/output && ' . $buildHook);
+    }
```

Comment:
Thanks for the PR! If I'm not mistaken everything above this is duplicated from the `deploy` command? Maybe that code could be shared instead, moved to a class maybe? I'm thinking a `generate()` method in a  `src/Console/Deployer` class for example.

Reply:
Yes you are right, it's a big duplication of deploy command. I would just propose you this feature in a simple way to saw if it's was a good idea.
I will refacto this and send you a new commit